apps/frontend/src/libs/components/search/libs/hooks/use-search-query-parameter.hook.ts

```diff
@@ -0,0 +1,43 @@
+import { useLocation, useNavigate } from "react-router-dom";
+
+import { useCallback } from "~/libs/hooks/hooks.js";
+
+import { SEARCH_PARAMETER_KEY } from "../constants/constants.js";
+
+type Properties = {
+	searchQuery: string;
+	updateSearchParams: (newSearchValue: string) => void;
+};
+
+const useSearchQueryParameter = (defaultSearch: string): Properties => {
```

Comment:
It should be consistent with the usePagination hook. We need to take values from query string inside it and the whole logic should be the same as for pagination. Also don't forget to keep the same naming and code style across the app
```suggestion
const useSearch = (): {
  search: string;
  onSearchChange: (search: string) => void;
} => {
```